bin/oneoff/populate_user_profile_csp

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env ruby
+
+# A oneoff script to import data from a CSV into the user_profiles table.
+#
+#   USAGE: ./bin/oneoff/populate_user_profile_csp /tmp/user_profile.csv
+#
+# The CSV is expected to be in the following format.
+#
+#   email,other_emails,other_user_ids,facilitator,nmsi,teals,pd_manual
+#   <email>,<other_emails>,<other_user_ids>,""|"true",""|"true",""|"true",""|"201x-201y"
+#   <email>,<other_emails>,<other_user_ids>,""|"true",""|"true",""|"true",""|"201x-201y"
+#   <email>,<other_emails>,<other_user_ids>,""|"true",""|"true",""|"true",""|"201x-201y"
+#
+# Note that <other_emails> and <other_user_ids> need to be escaped with
+# quotation marks if they contain commas (to separate multiple emails or IDs).
+
+require 'csv'
+require_relative '../../dashboard/config/environment'
+
+# @param filename [String] the path to the CSV file to read. The format of the
+#   file is described in the file comment above.
+# @return [Array[Hash]] each array element gives the data associated with the
+#   UserProfile to create.
+def read_csv(filename)
+  data = []
+  CSV.foreach(filename, {headers: true, header_converters: :symbol}) do |csv|
+    data << csv
+  end
+  return data
+end
+
+# param data [Array[Hash]] the UserProfiles to create.
+def process_data(data)
+  time_now = DateTime.now
+
+  data.each do |datum|
+    user = User.with_deleted.find_by_email_or_hashed_email(datum[:email])
+    if user.nil?
+      puts "MISSING USER: #{datum[:email]}"
+      next
+    end
+    if user.user_type == 'student'
+      puts "STUDENT USER: #{datum[:email]}"
+      next
+    end
+
+    user_profile = UserProfile.find_or_create_by(
+      user: user,
+      course: UserProfile::CSP
```

Comment:
At the moment, the priority is to understand CSP PD'ed teachers. At a later time, I expect to create a `populate_user_profile_zzz` script that populates the other courses.